AdminDashboard/BusinessLogicOrchestrators/AccountOrchestrator/AccountOrchestrator.cs

```diff
@@ -0,0 +1,125 @@
+﻿using AdminDashboard.Exceptions;
+using AdminDashboard.Main.Enumerations;
+using AdminDashboard.Models.Domain;
+using AdminDashboard.Models.EndUserRequests;
+using AdminDashboard.Models.JsonRequests;
+using AdminDashboard.Models.Plans;
+using AdminDashboard.Repositories.AccountRepository;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace AdminDashboard.BusinessLogicOrchestrators.AccountOrchestrator
+{
+    public class AccountOrchestrator : IAccountOrchestrator
+    {
+        private readonly IAccountRepository Repository;
+
+        public AccountOrchestrator(IAccountRepository repository)
+        {
+            Repository = repository;
+        }
+
+        public async Task RegisterUserToAccount(Guid accountId, Guid userId)
+        {
+            var account = await Repository.GetAccountBy(accountId);
+
+            if (DoesntExist(account))
+                throw new AccountNotFoundException();
+
+            if (HasUser(userId, account))
+                throw new DuplicateUserOnAccountException();
+
+            if (account.Users.Count == account.CurrentPlan.MaxNumberOfUsersAllowed)
+                throw new UserLimitExceededException();
+
+            await Repository.AddUser(accountId, userId);
+        }
+
+        public async Task UpgradePlan(Guid accountId, UpgradePlanRequest request)
+        {
+            var account = await Repository.GetAccountBy(accountId);
+
+            if (DoesntExist(account))
+                throw new AccountNotFoundException();
+
+            if (!IsProposedPlanHigherThanCurrentPlan(request.Plan, account))
```

Comment:
As far as I can tell, this is the only occurrence of `IsProposedPlanHigherThanCurrentPlan`, so why not call it `IsProposedPlanLessThanCurrentPlan` or similar and remove the negation?

Reply:
That's true, double negatives can be hard to intuit